kext/FileNVRAM/FileNVRAM.cpp

```diff
@@ -217,7 +217,7 @@ void FileNVRAM::copyUnserialzedData(const char* prefix, OSDictionary* dict)
             
             if(prefix)
             {
-                size_t size = strlen(prefix) + sizeof(NVRAM_SEPERATOR) + strlen(key->getCStringNoCopy());
+                size_t size = strlen(prefix) + sizeof(NVRAM_SEPERATOR) + strlen(key->getCStringNoCopy()) +1;
```

Comment:
I confirm that `strlen(NVRAM_SEPERATOR) +1` works better, no more bug, committed on my fork.
Now I'm going to add a notification (using addMatchingNotification) to match IOMediaBSDClient instead of using the timeout waiting for IOBSD that is clearly not working at least in newer OSes. Don't know if this is supported by the old Tiger's kernel and I need to do some research before doing this (or doing this based on kernel version).
(Anyway now we have the Pike's safe read/write operations that allow the kext to never crash or doing damages. I prefear datas to be restored by the bootloader so that the read operation is just an optional never used and also the module should be fixed due to a bug in xml.c)